acir/src/circuit/opcodes.rs

```diff
@@ -16,21 +16,25 @@ pub use oracle_data::OracleData;
 #[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub enum Opcode {
     Arithmetic(Expression),
+    /// Calls to "gadgets" which rely on backends implementing support for specialized constraints.
+    ///
+    /// Often used for exposing more efficient implementations of SNARK-unfriendly computations.  
     BlackBoxFuncCall(BlackBoxFuncCall),
     Directive(Directive),
     /// Abstract read/write operations on a block of data. In particular;
-    /// It does not require an initialisation phase
-    /// Operations do not need to be constant, they can be any expression which resolves to 0 or 1.
+    /// - It does not require an initialization phase
+    /// - Operations do not need to be constant, they can be any expression which resolves to 0 or 1.
     Block(MemoryBlock),
-    /// Same as Block, but it starts with an initialisation phase and then have only read operation
+    /// Same as Block, but it starts with an initialization phase and then have only read operation
     /// - init: write operations with index from 0..MemoryBlock.len
     /// - after MemoryBlock.len; all operations are read
+    ///
     /// ROM can be more efficiently handled because we do not need to check for the operation value (which is always 0).
     ROM(MemoryBlock),
     /// Same as ROM, but can have read or write operations
     /// - init = write operations with index 0..MemoryBlock.len
     /// - after MemoryBlock.len, all operations are constant expressions (0 or 1)
-    /// RAM is required for Aztec Backend as dynamic memory implementation in Barrentenberg requires an intialisation phase and can only handle constant values for operations.
+    // RAM is required for acvm-backend-barretenberg as dynamic memory implementation in Barretenberg requires an initialization phase and can only handle constant values for operations.
```

Comment:
I think we can, though it being here is a part of a larger discussion where we consider whether we need to support RAM/ROM + Block, or whether Block can be the fallback implementation for RAM/ROM and thats all it is.

Reply:
I think we can remove it as we shouldn't be mentioning barretenberg. But perhaps we should still mention that RAM operations only handle constant values

Reply:
>  whether we need to support RAM/ROM + Block, or whether Block can be the fallback implementation of RAM/ROM

Yeah, the status around the `Block` opcode isn't clear to me at all. https://github.com/noir-lang/noir/pull/1282 implies that order of preference is ROM > Block > RAM > permutations (which somehow still creates a Block opcode despite it being listed as unsupported!?). 

https://github.com/noir-lang/acvm-backend-barretenberg/blob/f6134b7b502cb74882300b0046ab91ab000daf3c/src/barretenberg_structures.rs#L1103-L1111 implies however that Block never gets passed to the backend at all (and so doesn't get converted into constraints) which seems preferable.

Very strange things happening without any documentation.

Reply:
Looking to merge this, @TomAFrench do you want to keep the Barretenberg reference and open an issue, so that we eventually remove when we review dynamic memory?

Reply:
Sure thing